Specifications/QIR/Data-Types.md

```diff
@@ -168,55 +139,48 @@ big integers.
 | __quantum__rt__bigint_greater     | `i1(%BigInt*, %BigInt*)`       | Returns true if the first big integer is greater than the second, false otherwise. |
 | __quantum__rt__bigint_greater_eq  | `i1(%BigInt*, %BigInt*)`       | Returns true if the first big integer is greater than or equal to the second, false otherwise. |
 
-### Tuples and User-Defined Types
-
-Tuple data, including values of user-defined types, is represented as an
-LLVM structure type.
-The structure type will contain a reference count as a standard header,
-followed by the tuple fields.
+## Tuples and User-Defined Types
 
-Because the `%TupleHeader` type actually appears as part of the structure, LLVM will
-not allow it to be an opaque type.
-Thus, QIR defines an LLVM type for the tuple header:
+Tuple data, including values of user-defined types, is represented as the corresponding LLVM structure type.
+For instance, a tuple containing two integers, `(Int, Int)`, would be represented in LLVM as `type {i64, i64}`.
 
-```LLVM
-%TupleHeader = type { i32 }
-```
-
-For instance, a tuple containing two integers, `(Int, Int)`, would be represented in
-LLVM as `type {%TupleHeader, i64, i64}`.
+When [invoking callable values](Callables.md#invoking-a-callable-value) using the `__quantum__rt__callable_invoke` runtime function, 
+tuples are passed as a pointer to an opaque LLVM structure, `%Tuple`. The pointer is expected to point to the contained data such that it can be cast to the correct data structures by the
+receiving code.
+This permits the definition of runtime functions that are common for all tuples, such as the functions listed below.
 
-Tuple handles are simple pointers to the underlying data structure.
-When passed to a callable function, tuples should always be passed by reference,
-even if the source language treats them as immutable.
+Many languages provide immutable tuples, along with operators that allow a modified copy of an existing tuple to be created.
+QIR supports this by requiring the runtime to track and be able to access the following given a `%Tuple*`:
+- The size of the tuple in bytes
+- The alias count indicating how many handles to the tuple exist in the source code
 
-In some situations, a pointer to a tuple of unknown or variable type may be required.
-To satisfy LLVM's strong typing in such a situation, tuple pointers are passed as
-pointers to the initial tuple header field; that is, as a `%TupleHeader*`.
-These pointers should be cast to pointers to the correct data structures by the
-receiving code.
-For instance, this convention is used for callable wrapper functions; see
-[below](#callable-values-and-wrapper-functions).
+The language specific compiler is responsible for injecting calls to increase and decrease the alias count as needed, as well as to accurately reflect when references to the LLVM structure representing a tuple are created and removed. 
```

Comment:
Yes, that is correct. However, to avoid the impression that QIR is in any way specific to Q# - which it is explicitly not meant to be - I'd rather avoid mention Q# over any other language that may choose to use this as their IR.